include/mm/pmm.h

```diff
@@ -100,6 +107,16 @@ static inline frame_t *get_first_frame(list_head_t *frames, unsigned int order)
     return list_first_entry(&frames[order], frame_t, list);
 }
 
+static inline bool frame_has_paddr(const frame_t *frame, paddr_t pa) {
+    if (!frame)
+        return false;
+
+    paddr_t start_pa = mfn_to_paddr(frame->mfn);
+    paddr_t end_pa = start_pa + ORDER_TO_SIZE(frame->order);
+
+    return pa >= start_pa && pa < end_pa;
```

Comment:
Should this be below instead to handle the very last pa (and unsigned overflow) as well?:
```c
    paddr_t end_pa = start_pa + ORDER_TO_SIZE(frame->order) - 1;

    return pa >= start_pa && pa <= end_pa;
```

Reply:
Fixed. Thanks.